### DISCUSSION TITLE
Enable azure tags in deployment validation scripts

### DISCUSSION THREAD

**Discussion by ThojoUno** [Azure/ResourceModules#3489] [Ideas]
We have a very restricted lab/test environment that REQUIRES Azure tags on all resource groups. Because of this, we can't perform deployment validation with the current branch. It would be quite easy to implement but requires updating every main.test.bicep template.
Steps to implement:
1. update ./settings.yml, add tags environment variable

Update Deployment validation section
location: 'centralus'
tags: '@{"Environment"="Lab";"Expiry Date"="12/31/2023";"Business Unit"="Cloud";"Owner"="Joe Thompson"}'

2. update settings.yml under .github/actions/templates/validateModuleDeployment

Replace AdditionalParameters = @{} with AdditionalParameters = @{ tags=${{ env.tags }} }
There are 2 instances in this file to replace.

3. update all .test/scenario/main.test.bicep files to add tags parameter for resource group resource..

Add 
@description('Optional. Some environments require tagging.')
param tags object = {}

Update
// General resources
// =================
resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
  name: resourceGroupName
  location: location
  tags: tags       //<=== add tag parameter
}

**Comment by eriqua** [comment]
Hi @ThojoUno, thanks for opening this discussion and sorry for the late reply. 

CARML evolved to and has been rebranded as the Bicep version of Azure Verified Modules ([AVM](https://aka.ms/AVM)). 

All modules previously hosted by this repo have been updated to align with AVM specifications and published to the Public Bicep Registry. Their evolved source code can be found in the underlying repository ([BRM](https://aka.ms/BRM)).

However, in order not to lose relevant discussions like this one, all CARML GitHub discussions are being converted to CARML GitHub issues, in order for the team to be able to further triage them and migrate them to AVM issues, if applicable.

Please refer to issue #4546 going forward.
